test/pass_files/alias_assign.d

```diff
@@ -0,0 +1,14 @@
+template T(X)
+{
+     alias A = int;
+     A = X;
+     struct B
+     {
+          A t;   
+     }
+}
+
+void fun()
+{
+    auto b = T!float.B();
+}
```

Comment:
really weak test, add more complex nesting, conditional compilation, static foreach, etc. what would be used with this feature.

Reply:
Fair point. I just wanted to get this out there as fast as possible as to get feedback.